sumologic/sumologic.py

```diff
@@ -40,7 +41,11 @@ def _get_endpoint(self):
         """
         self.endpoint = 'https://api.sumologic.com/api/v1'
         self.response = self.session.get('https://api.sumologic.com/api/v1/collectors')  # Dummy call to get endpoint
-        endpoint = self.response.url.replace('/collectors', '')  # dirty hack to sanitise URI and retain domain
+        if self.response.status_code == 401:
+            self.response = self.session.get('https://api.sumologic.com/api/v1/collectors')
+            endpoint = self.response.url.replace('/collectors', '')  # dirty hack to sanitise URI and retain domain
+        else:
+            endpoint = self.response.url.replace('/collectors', '')  # dirty hack to sanitise URI and retain domain
```

Comment:
That's unnecessary code duplicaiton.

Reply:
Thanks for catching it. Updated code to remove it.

Reply:
Actually, reviewing the recent commit logs. This can be closed.